docs/source/contributor-guide/index.md

```diff
@@ -72,13 +72,35 @@ recently active community members in the ticket to get their attention.
 
 ### What Features are Good Fits for DataFusion?
 
-DataFusion is designed to highly extensible, and many features can be implemented
-as extensions without changing the core of DataFusion.
-
-We are [working on criteria for what features are good fits for DataFusion], and
-will update this section when we have more to share.
-
-[working on criteria for what features are good fits for datafusion]: https://github.com/apache/datafusion/issues/12357
+DataFusion is designed to highly extensible, and many features can be
+implemented as extensions without changing the core of DataFusion. Support for
+new functions, data formats, and similar functionality can be added using those
+extension APIs, and there are already many existing community supported
+extensions listed in the [extensions list].
+
+Query engines are complex pieces of software to develop and maintain. Given our
+limited maintenance bandwidth, we try to keep the DataFusion core as simple and
+focused as possible, while still satisfying the [design goal] of an easy to
+start initial experience.
+
+With that in mind, contributions that meet the following criteria are more likely
+to be accepted:
+
+1. Bug fixes for existing features
+2. Test coverage for existing features
+3. Documentation improvements / examples
+4. Performance improvements to existing features (with benchmarks)
+5. "Small" functional improvements to existing features
+6. Additional APIs for extending DataFusion's capabilities
+
+Contributions that likely require discussion prior to acceptance include:
+
+1. New functionality that is part of the "standard sql"
+2. New functions that aren't part of the "standard sql"
```

Comment:
Not sure I'm following, 🤦  looks like every related to "standard sql" requires discussion?

Reply:
I changed the wording like this so nothing was "required" but to give a hint that major new features are likely to involve more discussion

```
Contributions that will likely involve more discussion (see Discussing New 
Features above) prior to acceptance include:

1. Major new functionality (even if it is part of the "standard sql")
2. New functions, especially if they aren't part of "standard sql"
3. New data sources (e.g. support for Apache ORC)
```